source/fundamentals/typescript.txt

```diff
@@ -41,17 +41,21 @@ Any object type can extend the ``Document`` interface.
 For more information on object types, see the
 `TypeScript handbook <https://www.typescriptlang.org/docs/handbook/2/objects.html>`__.
 
-Extend Document
-~~~~~~~~~~~~~~~
+Parameters that Extend Document
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The following classes accept any type that extends the ``Document``
-interface:
+The following classes accept all non-recursive types that extend
+the ``Document`` interface:
 
 - `Collection <{+api+}/classes/Collection.html>`__
 - `ChangeStream <{+api+}/classes/ChangeStream.html>`__
 
+To view an example of a recursive type, see the :ref:`<node-driver-limitations>` section.
```

Comment:
When reading through this time, I wonder if my comment during our discussion about ordering information might apply here. You're mentioning an example of something that won't work before you show an example of a type/interface that does work.